### DISCUSSION TITLE
Using "Usd Selection Kind"

### DISCUSSION THREAD

**Discussion by Aquamenti9** [Autodesk/maya-usd#3067] [Q&A] [answered]
Hello,

I have trouble with the Kind selection in maya 2024,
from what I understand of the documentation setting "usdSelectionKind" to "group" should result in selecting the first parent with kind=group if I select a prim in viewport that has kind=component (for example).

But I can't seem to get that result (it still select the component).
![Recording 2023-05-08 at 18 03 57](https://user-images.githubusercontent.com/102827498/236873371-492c8cf4-6c62-46f4-b94b-bcfe75c9b817.gif)

The gif from the documentation (https://help.autodesk.com/view/MAYAUL/2024/ENU/?guid=GUID-532E99C9-F638-49E3-9483-800FDB9B65D7) does not seem to show this result either, am I understanding the purpose of Kind selection wrong?
Is there something else to do to select the parent group of a component/subcomponent?

Thanks in advance!

**Comment by wallworm** [comment]
I believe this is a defect. I've brought it up with the team.

**Comment by santosd** [comment, marked as answer]
Hi @Aquamenti9, Could you try changing the value through the menu or through the following MEL command:
`mayaUsdSelectKindGroup;`
`mayaUsdSelectKindComponent;`
I tested this out on my end and it seems to be working as expected. When I try to query the "UsdSelecttionKind" option that you are querying on your end I get a value of 0 and I also used the -exists flag and get a return of 0 which tells me that is not the optionVar to set for kind. It seems that you might have created that optionVar, and set it's value but it is not actually linked to USD kind. I have attached a video that shows how I am setting the value on my end. I hope it helps.

https://github.com/Autodesk/maya-usd/assets/29235905/59e92520-2da7-49cd-b667-af87fd045426

**Reply by Aquamenti9** [reply]
Hello @santosd , thank you for your answer!
It does work as intended with the command you used :)
